core/src/main/java/jenkins/monitor/JavaVersionRecommendationAdminMonitor.java

```diff
@@ -46,15 +54,68 @@
 @Symbol("javaVersionRecommendation")
 public class JavaVersionRecommendationAdminMonitor extends AdministrativeMonitor {
 
+    /**
+     * The list of supported Java long-term support (LTS) releases. The key is the {@link
+     * Runtime.Version#feature() feature-release counter}. The value is the date the Jenkins project
+     * drops support for that release, which must be before the date the Eclipse Temurin project
+     * drops support for that release. This list must remain synchronized with the one in {@code
+     * executable.Main}.
```

Comment:
This could be asserted in a test, e.g.

```
    @Test
    void syncMonitor() {
        final Set<Integer> versions = JavaVersionRecommendationAdminMonitor.SUPPORTED_JAVA_VERSIONS.keySet();
        versions.forEach(it -> Main.verifyJavaVersion(it, false));
        final int mainVersionsCount = Main.SUPPORTED_JAVA_VERSIONS.size();
        if (versions.size() != mainVersionsCount) {
            throw new AssertionError("Expected version lists to match, but monitor has " + versions.size() + " and Main has " + mainVersionsCount);
        }
    }
```

(Plus required visibility changes)

Reply:
There is a reason I did not write a test in my original change. This cannot be made visible to `war/src/test` without also making it visible to `war/src/main`. I am not going to increase core API surface area in this PR, so rejecting this suggestion.